frontend/model/settings/vuexModule.js

```diff
@@ -70,17 +70,20 @@ const mutations = {
     state.increasedContrast = isChecked
   },
   setNotificationEnabled (state, enabled) {
-    if (state.notificationEnabled !== enabled) {
-      // We do this call to `service-worker` here to avoid DRY violations.
-      // The intent is creating a subscription if none exists and letting the
-      // server know of the subscription
-      sbp('service-worker/setup-push-subscription').catch(e => {
-        // The parent `if` branch should prevent infinite loops
-        sbp('state/vuex/commit', 'setNotificationEnabled', false)
-        console.error('[setNotificationEnabled] Error calling service-worker/setup-push-subscription', e)
-      })
-    }
+    console.info('[setNotificationEnabled] set to:', enabled)
     state.notificationEnabled = enabled
+    // We do this call to `service-worker` here to avoid DRY violations.
+    // The intent is creating a subscription if none exists and letting the
+    // server know of the subscription.
+    // The parent `if` branch should prevent infinite loops
+    sbp('service-worker/setup-push-subscription').catch(e => {
+      console.error('[setNotificationEnabled] Error calling service-worker/setup-push-subscription', e)
+      // if we attempted to turn it on and failed, then turn it off (so the user sees that the toggle
+      // switch is off and attempts to turn it back on again)
+      if (enabled) {
+        setTimeout(() => sbp('state/vuex/commit', 'setNotificationEnabled', false), 100)
```

Comment:
But:

1. Why do we need a timeout at all?
2. If the notification is not changing, you're calling `service-worker/setup-push-subscription` unnecessarily

Reply:
1. So that `setNotificationEnabled` is called again after this invocation and is set to `false` after it has been set to `true` by this invocation. It's not a good idea to call a Vuex commit thing on itself inside of the commit that's being committed, so a `setTimeout` is used to let the current commit finish.
2. No it's necessary as this ensures that for example, if `state.notificationEnabled` is `true` because it was loaded from the Vuex state, that other calls to this in the code to set it to true (e.g. by `setupNativeNotificationsListeners()`) still trigger a call to `'service-worker/setup-push-subscription'`. I can add a comment though to explain this.

Reply:
If you switch the toggle in the notification settings, you'll call `state.notificationEnabled` twice. I think know what you're trying to do, but as mentioned in my other comment, you need have another check to track the preference separately.